src/core/actions/unitactions/Attack.java

```diff
@@ -22,20 +23,39 @@ public Unit getTarget() {
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO compute all the attack actions for super.unit.
-        return null;
+
+        LinkedList<Action> attacks = new LinkedList<>();
+
+        if(isFeasible(gs)){
```

Comment:
Same as above, we wouldn't have a target set in this function. Instead, we need to return an action for each possible target this unit can attack to. Note we have to check that potential unit targets are from a different tribe.